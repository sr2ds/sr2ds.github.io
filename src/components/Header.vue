<template>
  <div class="Header">
    <router-link class="backIcon" v-if="$route.name !== 'home'" to="/">Voltar</router-link>
    <h1 :class="$route.name !== 'home' ? '': 'home-spacing'">David Silva</h1>
    <a class="linkedinIcon" href="https://www.linkedin.com/in/srdavidsilva/" target="_blank">
      <img src="@/assets/linkedin.png" alt="logo linkedin" title="Perfil LinkedIn" />
    </a>
    
  </div>
</template>

<script>
export default {
  name: 'Header'
};
</script>

<style>

.Header {
  width: 98vw;
  display: flex;
  justify-content: space-between;
}

.home-spacing {
  margin-left: 1rem;
}

.Header .linkedinIcon,
.Header .backIcon {
  margin: 1.2rem;
}
.Header .linkedinIcon img {
  width: 35px;
  height: 35px;
  fill: white;
  background-color: white;
  border-radius: 5px;
}
</style>