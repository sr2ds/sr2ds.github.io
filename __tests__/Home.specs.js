import Header from '../src/components/Header.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

describe('pagina Home', () => {

  it('garante que header esteja como no snapshot', () => {

    const wrapper = shallowMount(Header, { localVue, router })
    expect(wrapper.html()).toMatchSnapshot()
  });

});