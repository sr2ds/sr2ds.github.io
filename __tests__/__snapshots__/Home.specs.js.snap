// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagina Home garante que header esteja como no snapshot 1`] = `
<div class="Header">
  <router-link-stub to="/" tag="a" ariacurrentvalue="page" event="click" class="backIcon">Voltar</router-link-stub>
  <h1 class="">David Silva</h1> <a href="https://www.linkedin.com/in/srdavidsilva/" target="_blank" class="linkedinIcon"><img src="@/assets/linkedin.png" alt="logo linkedin" title="Perfil LinkedIn"></a>
</div>
`;
